test/Engine/ProtoTest/Macroblock/MacroblockGeneratorTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+using ProtoCore;
+using ProtoCore.Runtime;
+using ProtoCore.AssociativeGraph;
+using ProtoScript.Runners;
+namespace ProtoTest.Macroblocks
+{
+    public class MacroblockGeneratorTests : ProtoTestBase
+    {
+        private Core core = null;
+        ProtoCore.DSASM.Executable dsExecutable = null;
+        private List<MacroBlock> generatedMacroblocks = null;
+        public override void Setup()
+        {
+            // Compile core
+            core = new Core(new Options());
+            core.Compilers.Add(ProtoCore.Language.kAssociative, new ProtoAssociative.Compiler(core));
+            core.Compilers.Add(ProtoCore.Language.kImperative, new ProtoImperative.Compiler(core));
+        }
+
+        [Test]
+        public void TestSequential01()
+        {
+            const string code =
+                @"
+    a = 1;
+";
+            CompileAndGenerateMacroblocks(code);
+
+            // Verify
+            VerifyMacroblockCount(1);
+            VerifyGraphNodesInMacroblock(0, 1);
```

Comment:
Verifies the number of graphnodes associated with the macroblock. Im open to suggestions for additional verification methods